### ISSUE TITLE
MetaMask wallet is not open source

### ISSUE THREAD

**Issue by Aspie96** [ethereum/ethereum-org-website#7201]
MetaMask is [listed](https://ethereum.org/en/wallets/find-wallet/) as open source, but it [no longer is](https://twitter.com/MetaMask/status/1296586344299294720?). The source code of the browser extention is available, but the license is not open source.

The wallets page uses the phrase "open source", not "source available", and it also uses the OSI logo as a symbol, so this should be fixed.

**Comment by vj-codes** [comment]
@lukassim hey can I work on this?

**Comment by samajammin** [comment]
Hey @Aspie96 - thanks for raising this!

Agree with you that while MM source code is publicly available, it's not open source.

Rather than setting MM's `open_source` to `false`, I suggest the approach we take here is to re-work the [`open_source`](https://github.com/ethereum/ethereum-org-website/blob/dev/src/data/wallets/wallet-data.ts#L22) wallet data field from a `boolean` to enum with 3 options, e.g.:
- open source
- source available
- closed source

I think it's worth differentiating source available vs. closed source. There's still tremendous value for users & developers to be able to inspect & audit the source code of a wallet vs. something that is completely closed source.

I welcome input on this idea from the community! cc @corwintines

**Comment by corwintines** [comment]
I agree with the enum solution to this. I still think having source code available sets MM apart from closed source. 

@Aspie96 and community, would also like to hear what you think.

**Comment by Aspie96** [comment]
Sorry for my late reply.

I agree!

Because open source software is also source available, the filter should have 3 modes:
- No filter
- Source available (also includes OSS).
- Only OSS.

**Comment by uday03meh** [comment]
Hi @samajammin, I would like to work on this issue.
Please assign it to me.

**Comment by GHkrishna** [comment]
Hello guys, Do we have an update on this issue? If possible I can start working on the issue by changing the data type for the flag and allowing the three values suggested by @samajammin. 

I think changing this property can also affect other flow. For eg. Sorting on the basis of opensource on [the wallet page]( https://ethereum.org/en/wallets/find-wallet/ )

I would argue to rather keep the flag as it is and add another parameter giving more info about the code visibility. Please suggest.

**Comment by wackerow** [comment]
Circling back, this does about to be an issue still. 

I went ahead and assigned @corwintines as we're currently in the midst of some changes to the wallet page. @corwintines please feel free to re-assign as you see fit, but if you could help take a look at our options here given the existing structure that would be appreciated!

**Comment by wackerow** [comment]
Took a quick look at this... we're in the middle of revamping the wallet's page, so this won't have progress until after that is completed to avoid scope creep on that task.

We're using booleans for all of these features / filter options right now, which would require a little reworking to get a non-binary option in here... We can easily assign values to this such as `open_source: "OPEN" | "AVAILABLE" | "CLOSED", but we don't have a clean way to parse through these options with the current setup.

@nloureiro Maybe we can chat about this at some point in the near future... ideally we have a separate indicator for wallets that have their source available, but not free to fork/use (different from the open source icon). These features are currently filtered using toggles, which work great for a boolean option, but less so for an "enum" style choice. How we'd go about having a "three-way-toggle" of some kinda can be discussed.

**Comment by corwintines** [comment]
Based on discussion with the team, we would like to do the following:

- Maintain the functionality of the toggle. If something is source available or open source, it will be shown when the open source filter is applied
- We add a new category `source available` for wallets. They can now be `open source`, `closed source`, or `source available`
- Change metamask to source available

**Comment by alavarello** [comment]
Hi! Is anyone currently working on this? If not, I’d be happy to take it on! Let me know. You can assign it to me

**Comment by wackerow** [comment]
@alavarello Hey, circling back... any interest in picking this up still? Do you know the latest with that repo?

I don't believe anything has been changed to support this break down.... cc: @nloureiro or @konopkja any thoughts on "Closed / Available / Open" breakdown for this indicator?

If the source code _isn't_ fully available, then we should just remove if we don't have designs, else could potentially just close this out.

**Comment by konopkja** [comment]
<img width="301" height="194" alt="Image" src="https://github.com/user-attachments/assets/4eb62ad7-0434-413e-9eee-d8b1cee954f8" />

I would call it: Open source, Public code, Private code.

But looking at outr page if the intent is from a security perspective then we actually dont need to differentiate between open source and public source no? 

We just need to be able to audit the integrity of the application for this to pass, but i see the point why "open source" would be misleading. In that case the easiest fix is just to rephrase it to "Public code"

**Comment by nloureiro** [comment]
Closing as resolved: `src/data/wallets/wallet-data.ts` now has `open_source: false` for MetaMask, so the find-wallet page reflects this correctly. Thanks for the report — it took a while, but the data was fixed.
